### ISSUE TITLE
Setting small target epsilon like 0.1 fails training

### ISSUE THREAD

**Issue by LinkToPast1900** [lxuechen/private-transformers#15]
Hi, @lxuechen 
I tried to set epsilon as 0.1 on SST-2, but it results in a large noise_multiplier:  20853.95 and fails the training where the accuracy is near 0.5
However, setting epsilon as 1 works well.
Any idea about this?

**Comment by lxuechen** [comment]
> I tried to set epsilon as 0.1 on SST-2, but it results in a large noise_multiplier: 20853.95 and fails the training where the accuracy is near 0.5 

The fact that the noise multiplier scales non-linearly with epsilon is well-known. Given that the privacy accounting used in the paper is tight (the version that composes trade-off functions numerically), there's not much that can be done apart from collecting more examples and using a larger dataset. 

Other hparam tweaks -- increasing batch size, learning rate, and reducing number of updates -- could help. But the issue is fundamental.

**Comment by lxuechen** [comment]
In addition, I'll say that the frequency of the adoption of an epsilon as small as 0.1 is not high in the private deep learning literature. 

An epsilon that small appears more when the dataset is huge (millions, or even tens/hundreds of millions), or the learning problem is simple (e.g., count queries, etc.).

**Comment by LinkToPast1900** [comment]
Thanks!
